tests/get_or_create/models.py

```diff
@@ -32,6 +32,14 @@ class Thing(models.Model):
     name = models.CharField(max_length=255)
     tags = models.ManyToManyField(Tag)
 
+    @property
+    def capitalized_name_property(self):
+        return self.name
+
+    @capitalized_name_property.setter
+    def capitalized_name_property(self, val):
+        self.name = val.capitalize()
```

Comment:
When I remove the setter from the code, I get the following ``TypeError``:
```
======================================================================
ERROR: test_get_or_create_with_model_property_defaults (get_or_create.tests.GetOrCreateTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib64/python3.6/unittest/case.py", line 59, in testPartExecutor
    yield
  File "/usr/lib64/python3.6/unittest/case.py", line 601, in run
    testMethod()
  File "/home/freitafr/dev/django/tests/get_or_create/tests.py", line 80, in test_get_or_create_with_model_property_defaults
    t, _ = Thing.objects.get_or_create(defaults={'capitalized_name_property': "annie"}, pk=1)
  File "/home/freitafr/dev/django/django/db/models/manager.py", line 82, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/home/freitafr/dev/django/django/db/models/query.py", line 452, in get_or_create
    return self._create_object_from_params(lookup, params)
  File "/home/freitafr/dev/django/django/db/models/query.py", line 484, in _create_object_from_params
    obj = self.create(**params)
  File "/home/freitafr/dev/django/django/db/models/query.py", line 378, in create
    obj = self.model(**kwargs)
  File "/home/freitafr/dev/django/django/db/models/base.py", line 484, in __init__
    raise TypeError("'%s' is an invalid keyword argument for this function" % list(kwargs)[0])
TypeError: 'capitalized_name_property' is an invalid keyword argument for this function
```

I think Django should verify that the field can be set, just like the model initialization does (see [ticket #25910](https://code.djangoproject.com/ticket/25910)).